app/providers/session-provider.tsx

```diff
@@ -1,15 +1,23 @@
 'use client'
 
-import { ReactNode, createContext } from 'react'
+import { ReactNode, createContext, useContext } from 'react'
+import { redirect } from 'next/navigation'
 
-export const SessionContext = createContext<string | undefined>(undefined)
+const SessionContext = createContext<string | undefined>(undefined)
 
+export const useUserUuid = (): string => {
+  const uuid = useContext(SessionContext)
+  if (typeof uuid !== 'string') {
+    redirect('/')
+  }
+  return uuid
+}
```

Comment:
Whenever they try get their UUID, they're supposed to be login.

Reply:
Redirect to top is correct? Maybe /signin ?

Reply:
Ohhh yes ...!!
Thank you !